### ISSUE TITLE
Failing to start a recipe

### ISSUE THREAD

**Issue by davoclavo** [OpenAgInitiative/openag_ui#124]
@aguadopd reported that recipes can't be started via the UI [on the forum](http://forum.openag.media.mit.edu/t/food-computer-was-unable-to-start-recipe/1111)

Apparently the UI is sending the following JSON payload when starting a recipe via [this bit of code](https://github.com/OpenAgInitiative/openag_ui/blob/1ca22dfe423a085cbbac55fd5ab0b35c216f36b9/scripts/app.js#L342):

 `{"recipe_id": "test_recipe"}`

However, it looks like the [API expects the JSON payload to be an array of arguments](https://github.com/OpenAgInitiative/openag_brain/blob/master/src/openag_brain/software_modules/api.py#L136-L143) that will be passed to the service call, such as this:

`["test_recipe"]`

Here is a `curl` example request that should work with the current API. 

```
curl -XPOST 'http://10.0.0.8:5984/_openag/api/0.0.1/service/environment_1/start_recipe' \
     -H "Content-Type: application/json" \
     -d '["test_recipe"]'
```

I am not that familiar with this codebase yet, but I'll try to submit a PR for this soon.

**Comment by gordonbrander** [comment]
@davoclavo let me know if you plan on working on this. It's on my list, so I'll probably make a fix soon if you don't get there first.

**Comment by davoclavo** [comment]
@gordonbrander I don't want to side track too much, but what are your thoughts on ditching the API software module entirely and rely on [rosbridge_suite](http://wiki.ros.org/rosbridge_suite) instead, which runs a server that exposes a websocket interface to interact with all the ROS stuff (Params, Topics, Services). Then on the front end side use [roslibjs](http://wiki.ros.org/roslibjs/Tutorials/BasicRosFunctionality), which is a thin async JS wrapper to talk to such ws service.

Is there a specific reason to keep the API module as is? The only thing I can think of is figuring out is if it is possible to proxy the websockets via CouchDB.

Regardless, I think you'd be able to fix/test this faster than I would, given that I have no experience with this repo yet.

**Comment by gordonbrander** [comment]
@davoclavo we've discussed this, but it's been on the back-burner.

One requirement is that the ROS REST API (whatever powers it) must be able to proxy through CouchDB's server, since we plan to use Couch's authentication system to protect the API as well.

If you want to submit a PR for such a thing, that would be awesome!

cc @LeonChambers

**Comment by LeonChambers** [comment]
I'm very much in favor of switching to rosbridge as long as CouchDB can proxy it. I imagine this would require a good amount of re-writing on the frontend though. It's pretty easy in the backend.

**Comment by gordonbrander** [comment]
Assuming rosbridge can do REST/http as well as sockets, patching the UI should be easy. If it's sockets only, it might take a bit more work, but shouldn't be too hard.

The only thing we might want to watch out for is cross-origin restrictions with websockets. Right now the UI is able to talk with the bot through CORS. I'm not sure what the restrictions are on websockets.

**Comment by LeonChambers** [comment]
"While rosbridge server provides a WebSocket connection, rosbridge itself is not tied to any particular transport layer. You could add a rosbridge TCP package or node, for example, that communicates with rosbridge using TCP sockets. No WebSockets or browsers needed." (from [this page](http://wiki.ros.org/rosbridge_server)). So it looks like normal HTTP isn't supported.

CORS should still work fine since the initial handshake for WebSockets is actually just an HTTP request.

**Comment by davoclavo** [comment]
yeah, @LeonChambers is right. rosbridge won't work as long as Couchdb doesn't provide a WS or TCP proxy :(.  And I couldn't find a rosbridge HTTP interface either, so perhaps the API software module eventually become that HTTP interface! I bet other projects would use it as well.

Anyways, back to the issue. I got my dev environment setup for openag_ui and fixed that JSON payload by making it an array of values. However, if there is no recipe running when doing so (like on a fresh install) briefly prompts an error message because it always tries to *stop* the previous recipe and then *start* the next one. I tried fixing that as well, but I am still not able to get it to work. (Will continue the discussion in the PR.)
